### PR TITLE
First steps as tutorial. Data Sources sub-section done

### PR DESCRIPTION
**Pull request by emadehsan**
First steps" is now a tutorial that walks the read through the code by doing while making sure that they are exposed to enough *CliMetLab* APIs that they start to get comfortable and ready to use it.

It assumes not prior knowledge from the reader except basic Python programming. 
E.g. reader doesn't have to know about GRIB or NetCDF format.

Have not taken away any of previous content. Only modified and added new.

Your feedback and request for modifications are most welcome!

PS:
* The reason I added basic explanations for GRIB, NetCDF etc was so that any user (Python programmer) and not just a meteorologist could understand and get up to speed with CliMetLab

* I pushed the CSV download example to the middle instead of start because it is 34 MB file and takes a
while to download on a slower network (like mine). So a new reader can easily get distracted in the meantime

* Removed the links to example notebooks (they were giving 404 anyway) in the table of Data Sources section. 
As this can overwhelm the new user who is just getting started, with so many links to follow and read. 

* Tried to incorporate some important lessons from [What nobody tells you about documentation](https://www.youtube.com/watch?v=t4vKPhjcMZg)

### CODE REVIEW

**Comment by FussyDuck** [comment]
[CLA assistant check](http://claassistant.ecmwf.int:80/ecmwf/climetlab?pullRequest=32) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](http://claassistant.ecmwf.int:80/ecmwf/climetlab?pullRequest=32) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](http://claassistant.ecmwf.int:80/check/ecmwf/climetlab?pullRequest=32) it.</sub>

File: docs/firststeps.rst [hunk 1]
```diff
@@ -1,35 +1,129 @@
...
+from where we can download or access files related to climate data.
+
+To get started, let us first import *CliMetLab* at the top of our Python notebook:
+
+.. code-block:: python
+
+    import climetlab as cml
+
+Now download
+`test.grib <https://raw.githubusercontent.com/ecmwf/climetlab/develop/docs/examples/test.grib>`_ (example GRIB file)
```
Comment (floriankrb) [inline docs/firststeps.rst:31]: The example grib test files is https://get.ecmwf.int/repository/test-data/climetlab/test.grib.
You can browse https://get.ecmwf.int/service/rest/repository/browse/test-data/climetlab/ to see more.

File: docs/firststeps.rst [hunk 2]
```diff
@@ -1,35 +1,129 @@
...
+Here we used ``load_source`` method from *CliMetLab* to load our GRIB file
+into ``grib_data`` variable. The first argument ``"file"`` specifies the type of
+our data source. Which currently is indeed a local file that we downloaded.
+The second argument is the path to that file.
+
+Let's plot this data using the ``plot_map`` convenience method:
+
+.. code-block:: python
+
+    cml.plot_map(data, title=True, path="test-grib-plot.svg")
```
Comment (floriankrb) [inline docs/firststeps.rst:62]: "data" was called "grib_data" before.

File: docs/firststeps.rst [hunk 3]
```diff
@@ -1,35 +1,129 @@
...
+.. code-block:: python
+
+    netcdf_url = "https://raw.githubusercontent.com/ecmwf/climetlab/develop/docs/examples/test.nc"
+    netcdf_data = cml.load_source("url", netcdf_url)
+    cml.plot_map(netcdf_data)
+
+.. image:: _static/example-plots/test-netcdf-plot.svg
+  :width: 100%
+
+If you don't know already, NetCDF is another commonly used data format for
```
Comment (floriankrb) [inline docs/firststeps.rst:86]: Add a link to netcdf, maybe?

File: docs/firststeps.rst [hunk 4]
```diff
@@ -1,35 +1,129 @@
...
+transporting scientific data.
+
+In this case, *CliMetLab* not only downloaded the file specified by the url,
+but it also saved it in the
```
Comment (floriankrb) [inline docs/firststeps.rst:90]: We don't want to expose the cache location, especially for first users. Cache inspection and management should be done with the shell commands "climetlab cache ..." and "climetlab decache ..." (to be documented)

File: docs/firststeps.rst [hunk 5]
```diff
@@ -1,35 +1,129 @@
...
+Now download
+`test.grib <https://raw.githubusercontent.com/ecmwf/climetlab/develop/docs/examples/test.grib>`_ (example GRIB file)
+to your project directory, or if you have ``wget`` command available, run
+following in your notebook:
+
+.. code-block:: python
+
+    !wget https://raw.githubusercontent.com/ecmwf/climetlab/develop/docs/examples/test.grib
+
+GRIB is a file format for storage and sharing of gridded meteorological data.
```
Comment (floriankrb) [inline docs/firststeps.rst:39]: We do assume some previous knowledge of the domain, so describing GRIB is not necessary. But it is nice to have this for completeness.

File: docs/firststeps.rst [hunk 6]
```diff
@@ -1,35 +1,129 @@
...
+You can think of gridded data as weather or some other data that is associated with
+specific locations using coordinates. For example, wind speed data for
+every longitude and latitude on a two dimensional grid.
+
+GRIB is designed and maintained by World Meteorological Organization as an
+easy to handle and compact data format. GRIB (GRIdded Binary) is a binary file
+format so you cannot look at it using a text editor. But you sure can use
+*CliMetLab* to explore it:
```
Comment (floriankrb) [inline docs/firststeps.rst:47]: "designed and maintained by" ?
Perhaps instead use: "The GRIB format (version 1 and 2) is endorsed by WMO" with a link.

"easy to handle" is unnecessary IMHO.

**Review by floriankrb — CHANGES_REQUESTED** [review]
These changes are very welcome, I think the first step, using "load_source" is important and this is definitely easier to read. I made a few comments below.

File: docs/firststeps.rst (hunk 1 above)
Comment (emadehsan) [reply docs/firststeps.rst:31]: This GRIB file form https://get.ecmwf.int/repository/test-data/climetlab/test.grib
contains some content from HTTP response header 
```
--4b7944d0f72bc34f1e8aee521247db75
Content-Disposition: form-data; name="file"; filename="test.grib"

```

Same for the test.nc file from https://get.ecmwf.int/repository/test-data/climetlab/test.nc
```
--43d09ae0d3d0d9aaef6a38cedaac8135
Content-Disposition: form-data; name="file"; filename="test.nc"
.
.
.
--43d09ae0d3d0d9aaef6a38cedaac8135--
```

For this reason, `load_source` is not able to read these files as GRIB or NetCDF. May be the HTTP Server is adding these headers to the contents of these files

File: docs/firststeps.rst (hunk 3 above)
Comment (emadehsan) [reply docs/firststeps.rst:86]: A [Wikipedia](https://en.wikipedia.org/wiki/NetCDF) or link to [NetCDF docs](https://www.unidata.ucar.edu/software/netcdf/docs/)?

**Comment by emadehsan** [comment]
Hi Florian, I've made changes as you suggested. Please feel free to let me know if something else needs to be modified.

In the case of link to test.grib files from ECMWF servers, [there was an error](https://github.com/ecmwf/climetlab/pull/32/files#r752913892), so I didn't change those links.

File: docs/firststeps.rst (hunk 1 above)
Comment (emadehsan) [reply docs/firststeps.rst]: Ok. I've updated the links.

**Review by floriankrb — APPROVED** [review]
Thank you for this improvement!

**Pull request merged**